concordium-std/src/test_infrastructure.rs

```diff
@@ -1287,21 +1287,32 @@ pub struct TestHost<State> {
     /// Functions that mock responses to calls.
     // This is Rc+RefCell because it needs to be cloneable. There might be another way to make the
     // MockFn cloneable, but this seemed like the easiest option.
-    mocking_fns:      Rc<RefCell<MockFnMap<State>>>,
+    mocking_fns:             Rc<RefCell<MockFnMap<State>>>,
     /// Transfers the contract has made during its execution.
-    transfers:        RefCell<Vec<(AccountAddress, Amount)>>,
+    transfers:               RefCell<Vec<(AccountAddress, Amount)>>,
     /// The contract balance. This is updated during execution based on contract
     /// invocations, e.g., a successful transfer from the contract decreases it.
-    contract_balance: RefCell<Amount>,
+    contract_balance:        RefCell<Amount>,
+    /// The address of the instance. This is used when querying its own balance.
+    contract_address:        Option<ContractAddress>,
```

Comment:
I'm not exactly sure why this is necessary. I don't think there is special logic for this, is there?

Reply:
Resolving this to keep the discussion in https://github.com/Concordium/concordium-rust-smart-contracts/pull/185/files/6ef5176a4dc3f1e8b614132fd114c19d95de3a01#r1012234240